src/AppInstallerCLICore/Argument.cpp

```diff
@@ -145,17 +168,121 @@ namespace AppInstaller::CLI
         return strstr.str();
     }
 
-    void Argument::ValidatePackageSelectionArgumentSupplied(const Execution::Args& args)
+    void Argument::ValidateExclusiveArguments(const Execution::Args& args, const std::vector<Args::Type>& argTypes)
     {
-        for (Args::Type type : { Args::Type::Query, Args::Type::Manifest, Args::Type::Id, Args::Type::Name, Args::Type::Moniker, Args::Type::ProductCode, Args::Type::Tag, Args::Type::Command })
+        std::vector<Args::Type> argsUsed;
+        std::copy_if(argTypes.begin(), argTypes.end(), std::back_inserter(argsUsed), [&](Args::Type arg) { return args.Contains(arg); });
+
+        if (argsUsed.size() > 1)
         {
-            if (args.Contains(type))
+            // Create a string showing the exclusive args.
+            std::string argsString;
+            for (const auto& arg : argsUsed)
             {
-                return;
+                if (!argsString.empty())
+                {
+                    argsString += "|";
+                }
+
+                argsString += GetArgumentName(arg);
             }
+
+            throw CommandException(Resource::String::MultipleExclusiveArgumentsProvided(Utility::LocIndString{ argsString }));
+        }
+    }
+
+    ArgTypeCategory Argument::GetCategoriesPresent(const Execution::Args& args)
+    {
+        ArgTypeCategory result = ArgTypeCategory::None;
+
+        WI_SetFlagIf(result, ArgTypeCategory::Manifest, args.Contains(Args::Type::Manifest));
+
+        if (ContainsArgumentFromList(args, {
+            Args::Type::Exact,
+            }))
+        {
+            WI_SetAllFlags(result, ArgTypeCategory::PackageQuery | ArgTypeCategory::SinglePackageQuery);
+        }
+
+        WI_SetFlagIf(result, ArgTypeCategory::SinglePackageQuery, ContainsArgumentFromList(args, {
+            Args::Type::Query,
+            Args::Type::Id,
+            Args::Type::Name,
+            Args::Type::Moniker,
+            Args::Type::ProductCode,
+            Args::Type::Tag,
+            Args::Type::Command,
+            Args::Type::Version,
+            Args::Type::Channel,

```

Comment:
Technically couldn't `Tag` or `Command` also?
```
PS C:\WINDOWS\system32> winget search --tag chrome
Name                       Id                         Version      Match
----------------------------------------------------------------------------------------
Chrome Remote Desktop Host Google.ChromeRemoteDesktop 110.0.5481.7 Tag: chrome
Inssist                    SlashedIo.Inssist          16.1.0       Tag: chrome-extension

PS C:\WINDOWS\system32> winget search --command pdf
Name         Id                    Version  Match
----------------------------------------------------------------
wkhtmltox    wkhtmltopdf.wkhtmltox 0.12.6-1 Command: wkhtmltopdf
SumatraPDF   SumatraPDF.SumatraPDF 3.4.6    Command: sumatrapdf
PDFtk Server PDFLabs.PDFtk.Server  2.02     Command: pdftk
PDFtk Free   PDFLabs.PDFtk.Free    2.02     Command: PdftkXp
```

Reply:
I didn't think of that. Now I can see how channels could be somewhat standardized and not completely tied to each package, like "release" or "beta".

I can also see it for the tags, multiple packages sharing the same one, but not so much for the command. Command seems like you would only want one instance of each command, even though for search it looks for substring instead of full match.

Reply:
> Command seems like you would only want one instance of each command, even though for search it looks for substring instead of full match.

Even with `-e`, there are still multiple packages that use the same command. Visual studio is a great example. Even if channels are implemented to allow for the abstraction of Professional/Community/Enterprise, the different versions such as 2019/2022 use the same command -
```
PS C:\WINDOWS\system32> winget search --command devenv -e
Name                                    Id                                               Version             Match
----------------------------------------------------------------------------------------------------------------------------
Visual Studio Professional 2022 Preview Microsoft.VisualStudio.2022.Professional-Preview 17.0.0 Preview 2.1  Command: devenv
Visual Studio Professional 2022         Microsoft.VisualStudio.2022.Professional         17.4.4              Command: devenv
Visual Studio Enterprise 2022 Preview   Microsoft.VisualStudio.2022.Enterprise-Preview   17.0.0 Preview 2.1  Command: devenv
Visual Studio Community 2022 Preview    Microsoft.VisualStudio.2022.Community-Preview    17.0.0 Preview 2.1  Command: devenv
Visual Studio Professional 2019 Preview Microsoft.VisualStudio.2019.Professional-Preview 16.11.0 Preview 3.0 Command: devenv
Visual Studio Professional 2019         Microsoft.VisualStudio.2019.Professional         16.11.23            Command: devenv
Visual Studio Enterprise 2019 Preview   Microsoft.VisualStudio.2019.Enterprise-Preview   16.11.0 Preview 3.0 Command: devenv
Visual Studio Community 2019 Preview    Microsoft.VisualStudio.2019.Community-Preview    16.11.0 Preview 3.0 Command: devenv
Visual Studio Community 2022            Microsoft.VisualStudio.2022.Community            17.4.4              Command: devenv
Visual Studio Team Explorer 2022        Microsoft.VisualStudio.2022.TeamExplorer         17.4.4              Command: devenv
Visual Studio Load Test Controller 2022 Microsoft.VisualStudio.2022.TestController       17.4.4              Command: devenv
Visual Studio Community 2019            Microsoft.VisualStudio.2019.Community            16.11.23            Command: devenv
Visual Studio Team Explorer 2019        Microsoft.VisualStudio.2019.TeamExplorer         16.11.23            Command: devenv
Visual Studio Load Test Controller 2019 Microsoft.VisualStudio.2019.TestController       16.11.23            Command: devenv
```

Reply:
I think we would categorize them based on user intent (Command seems like it's for single package). Or just put it in PackageQuery if we are not sure. The previous code just tries to validate that --manifest and PackageQuery should not exist together. Either way I'm not too worried.